tools/pihole_toml_to_markdown.py

```diff
@@ -0,0 +1,192 @@
+import re
+from pathlib import Path
+
+def parse_toml_with_comments(filepath):
+    """
+    Parse a TOML file with comments and generate a Markdown documentation string.
+
+    Args:
+        filepath (str or Path): Path to the TOML file.
+
+    Returns:
+        str: Markdown-formatted documentation.
+    """
+    with open(filepath, encoding='utf-8') as f:
+        lines = f.readlines()
+
+    # Start with a Markdown header and introductory documentation block
+    documentation = ["""<!-- markdownlint-disable MD033 -->
+# Pi-hole FTL Configuration Reference
+
+This page documents the available options of `pihole-FTL`. They are typically managed via the [TOML](https://toml.io/)-formatted configuration file `/etc/pihole/pihole.toml`. This file may be edited directly or you can use the command line (CLI) option, the web interface, the application programming interface (API) or environment variables.
+
+Using the web interface, the API or the CLI is preferred as they can do error checking for you, trying to prevent any incompatible options which could prevent FTL from starting on a severely broken configuration.
+
+To edit with the command line, use the format `key.name=value`, e.g:
+
+```text
+sudo pihole-FTL --config dns.dnssec=true
+```
+
+!!! note "Environment Variables"
+    **⚙️ Configuration Precedence**
+    Every Pi-hole setting in this file can be overridden using an environment variable.
+    This is especially common in Docker deployments.
+    Environment variable names follow the format:
+    ```text
+    FTLCONF_<section>_<key>
+    ```
+    For example:
+    ```text
+    FTLCONF_dns_upstreams
+    FTLCONF_database_DBimport
+    ```
+    ⚠️ **If a setting is defined via an environment variable, it becomes read-only.**
+    You will not be able to override it through the TOML file, the command line, or the web interface until the variable is removed from the environment.
+
+---
+<!-- markdownlint-disable-file MD034 -->
+"""]
+    section_stack = []      # Tracks the current TOML section(s)
+    comment_buffer = []     # Collects comments preceding a key
+    in_config = False       # True after the first section header
+
+    for line in lines:
+        stripped = line.strip()
+
+        # Section header: [section]
+        if re.match(r'^\[.*\]$', stripped):
+            in_config = True
+            section_stack = [stripped.strip('[]')]
+            documentation.append(f"## [{'.'.join(section_stack)}]\n")
+            continue
+
+        # Array of tables: [[section]]
+        elif re.match(r'^\[\[.*\]\]$', stripped):
+            in_config = True
+            section_stack = [stripped.strip('[]')]
+            documentation.append(f"## [{'.'.join(section_stack)}]\n")
+            continue
```

Comment:
This block is never used.

Our TOML file doesn't use sections with double brackets (`[[section]]`).

Reply:
Double square brackets isn't a thing in TOML at all for *sections*